components/style/properties/shorthand/position.mako.rs

```diff
@@ -512,7 +510,7 @@
             if *self.grid_template_areas != Either::Second(None_) ||
                (*self.grid_template_rows != GridTemplateComponent::None &&
                    *self.grid_template_columns != GridTemplateComponent::None) ||
-               self.is_initial() {
+               self.is_grid_template() {
```

Comment:
This test sounds somewhat hacky in general, do we have any reason for it apart from "is how gecko happens to implement"? If so, can we add a comment on this?